graphrag/index/update/dataframes.py

```diff
@@ -225,13 +242,21 @@ def _update_and_merge_relationships(
         The updated relationships.
     """
     # Increment the human readable id in b by the max of a
-    delta_relationships["human_readable_id"] += (
-        old_relationships["human_readable_id"].max() + 1
-    )
-
-    # Merge the final relationships
+    # Ensure both columns are integers
+    delta_relationships["human_readable_id"] = delta_relationships[
+        "human_readable_id"
+    ].astype(int)
+    old_relationships["human_readable_id"] = old_relationships[
+        "human_readable_id"
+    ].astype(int)
+
+    # Adjust delta_relationships IDs to be greater than any in old_relationships
+    max_old_id = old_relationships["human_readable_id"].max()
+    delta_relationships["human_readable_id"] += max_old_id + 1
```

Comment:
What about this?

```py
initial_id = old_relationships["human_readable_id"].max() + 1
delta_relationships["human_readable_id"] = np.arange(initial_id, initial_id+len(delta_relationships))
```

![{F3A10653-18D5-4891-B27A-04A1391D7301}](https://github.com/user-attachments/assets/84488ec3-aab1-4fcc-a7bb-93bd6de330af)